Shield Rework

CluckinAwesome opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request

Removes:
Goliath shield
Roman shield (real and fake)
Riot shield
Spiked riot shield
Flash riot shield

Adds:
New sprites for the telescopic shield and the brand new Heavy shield
<img width="583" height="262" alt="image" src="https://github.com/user-attachments/assets/ed1ff181-aa61-4f18-a8c3-860cd52948e3" />
Sprites were done by the amazing @thearbiber

The Heavy Shield
A shield, as the name implies, very heavy. Leading to a very, very significant slowdown for the user
With this weight comes a much higher block chance, a slightly higher max integrity compared to the now deceased riot shield, and a better ap threshold as well

The Telescopic Shield
Decreased the total integrity as well as the block chance, but also lightly decreased the slowdown

Shields now have their own respective repair material

Replaces the old riot shield in the Vaquero with a Heavy shield

<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

## Why It's Good For The Game

Removes shields that didn't belong with the setting (the roman shields), as well as shields that are practically never used in-game (and the flash-shield was a bit buggy with its use, from what I have been told from testing)

As for the changes made to the shields themselves, namely the telescopic and the new heavy shield

The telescopic shield was lightly adjusted to better fit as a "melee" focused shield. Not ideally used in firefights, but more for close-quarters-combat involving melee based weaponry.

The heavy shield on the opposite side is a lot better suited for firefights, but namely for simply taking light-to-medium gunfire. The main idea for the heavy shield is to enable people to act almost like "mobile defense", giving up most if not all of their offensive capability to lean heavy into defensive capability. If used with the simple one-hand, it is practically 1:1 to the old riot shield, with the only difference being more slowdown. If used with two-hands however the slowdown is increased significantly more as well as the block chance. Alongside the addition of weapons being able to penetrate shields now, I don't believe this should cause any serious balance issues.

As for the removal of the riot shield, it overlapped a lot with the implementation of the heavy shield. With the only real difference between the two the slowdown difference and the ability to wield the heavy shield. With this overlap I decided it best to simply remove the riot shield, and replace it with the heavy shield.

<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding. -->

## Changelog

:cl: Cluckinawesome, thearbiber
add: Added the Heavy Shield
del: Removed Goliath shield, Roman shield (real and fake), Riot shield, Spiked riot shield, Flash riot shield
balance: Lightly touched telescopic shield's stats to make it a little weaker
imageadd: updated the sprites used for the telescopic shield and the new sprites for the heavy shield
imagedel: deleted the Goliath shield, Roman shield (real and fake), Riot shield, Spiked riot shield, Flash riot shield,
code: Updated shields to not all use plasteel, each now has a respective material that is used in repair
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

CluckinAwesome committed:
Update shields.dm

CluckinAwesome committed:
Removed all mentions of roman, flash, and spike shield

CluckinAwesome committed:
losingmymind

CluckinAwesome committed:
Increased slowdown and max integrity

CluckinAwesome committed:
basic shield sprite update

CluckinAwesome committed:
further shield sprite updating help me

CluckinAwesome committed:
Merge remote-tracking branch 'upstream/master' into shieldUpdate

CluckinAwesome committed:
where were you when shield/riot is kill? Also forgot a sprite for telescopic

CluckinAwesome committed:
Merge remote-tracking branch 'upstream/master' into shieldUpdate

CluckinAwesome committed:
Merge remote-tracking branch 'upstream/master' into shieldUpdate

CluckinAwesome committed:
vintage story magnetite torment nexus (Fixed repair proc, removed goliath shield, and updated vaquero with new shield))

CluckinAwesome committed:
Merge remote-tracking branch 'upstream/master' into shieldUpdate

CluckinAwesome committed:
please let it be over (heavy shields should no longer block despite breaking)

CluckinAwesome committed:
missed a spot FUCK

CluckinAwesome committed:
Panic attack thinking I forgot to update cargo (I already did)

CluckinAwesome committed:
Can finally update the script name with the PR number

CluckinAwesome committed:
goliath shield came back to haunt me (replaced it with the buckler)

CluckinAwesome committed:
Please let this be the last repath update.. (Repathed bucklers and swapped tele shields for heavy shields for CMM police))

CluckinAwesome committed:
Please let this be the last repath (Updated telescopic shield path for the Talos)

LettuceGuardian commented:
You sure the roman shield should be removed? Perhaps changed into a useless, generic item for fluff and/or RP purposes. I can see a Mercury-Drua ship appreciating having more stuff to be anthropological/archeological nerds about.

meemofcourse commented:
romans did not discover space travel

LettuceGuardian commented:
Or did they..? Perhaps made enemies..?
![G5QO1s8XsAAGpaD-1](https://github.com/user-attachments/assets/226d8cf8-867f-4aee-81e7-05c8b77fceb3)

It's got funny RP implications.

thgvr commented:
i too saw this image on twitter

meemofcourse commented:
i appreciate the suggestion but it's just not tonally appropriate with the lore of the setting

CluckinAwesome committed:
Talos-be-gone

CluckinAwesome committed:
Merge branch 'master' into shieldUpdate

Erikafox commented on code/game/objects/items/shields.dm:
```diff
@@ -32,6 +34,14 @@
+	var/is_repairable = TRUE
+	// The block chance the shield will revert to when repaired
+	var/standard_block_chance = 60
+	// The slow down the shield will revert to when repaired
```
these don't properly document unless you do ///

Will properly review later.

CluckinAwesome replied:
Updated, thanks for the heads-up!

Erikafox committed:
Merge branch 'master' into shieldUpdate

CluckinAwesome committed:
Fixed comments for documentation (Thank you Erika)

CluckinAwesome committed:
Merge branch 'shieldUpdate' of https://github.com/CluckinAwesome/Shiptest into shieldUpdate

thgvr commented on code/game/objects/items/shields.dm:
```diff
@@ -1,7 +1,7 @@
 
 /obj/item/shield
-	name = "shield"
+	name = "Shield"
```
```suggestion
	name = "shield"
```
dunno what this was for

thgvr commented on code/game/objects/items/shields.dm:
```diff
@@ -326,18 +253,50 @@
+
+/// triggered on wield of two handed item
+/obj/item/shield/heavy/proc/on_wield(obj/item/source, mob/user)
+	SIGNAL_HANDLER
```
```suggestion
```
this shouldnt be necessary

thgvr reviewed (dismissed):
all of this just makes me wish we had a more standard system for repairing things

CluckinAwesome committed:
Update code/game/objects/items/shields.dm

Co-authored-by: thgvr <81882910+thgvr@users.noreply.github.com>
Signed-off-by: CluckinAwesome <158237114+CluckinAwesome@users.noreply.github.com>

CluckinAwesome committed:
Update code/game/objects/items/shields.dm

Co-authored-by: thgvr <81882910+thgvr@users.noreply.github.com>
Signed-off-by: CluckinAwesome <158237114+CluckinAwesome@users.noreply.github.com>

generalthrax commented:
If you wield again while the progress bar to wield is ongoing the off-hand item is forever stuck to your hand

CluckinAwesome committed:
Lava Canyon is dead

CluckinAwesome committed:
Merge branch 'master' into shieldUpdate

CluckinAwesome committed:
Merge remote-tracking branch 'upstream/master' into shieldUpdate

CluckinAwesome committed:
Fixed sprites, lightly lowered slowdown, fixed path updates

CluckinAwesome committed:
The pixels didn't like me

CluckinAwesome committed:
Merge branch 'shieldUpdate' of https://github.com/CluckinAwesome/Shiptest into shieldUpdate

KOCMODECAHTHUK commented on code/game/objects/items/shields.dm:
```diff
@@ -77,164 +88,72 @@
+				broken = FALSE
+				block_chance = standard_block_chance
+				slowdown = standard_slowdown
+				drag_slowdown = standard_slowdown
```
Not to sound harsh, but I literally laughed at making a separate variable that duplicates an existing one.
```suggestion
				block_chance = initial(block_chance)
				slowdown = initial(slowdown)
				drag_slowdown = initial(drag_slowdown)
```

FalloutFalcon replied:
What @KOCMODECAHTHUK said is correct, you should use initial or :: here. However he needs to carry himself better (Do not talk about laughing about mistakes a contributer makes... We all make them dude..).
(You already created a weird issue where the shield starts at 50 but repairing it boosts it to 60)

KOCMODECAHTHUK replied:
Yeah, I just suggested a better way this moment.
I already wrote above that I didn't want to seem rude.

Erikafox commented:
<img width="168" height="26" alt="image" src="https://github.com/user-attachments/assets/6d4e995d-b8c6-4d5f-b668-a49117e73bbd" />
Either remove or fluff the block chance description blurb

CluckinAwesome commented:
> <img alt="image" width="168" height="26" src="https://private-user-images.githubusercontent.com/94164348/532991091-6d4e995d-b8c6-4d5f-b668-a49117e73bbd.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Njc5MjEzMTgsIm5iZiI6MTc2NzkyMTAxOCwicGF0aCI6Ii85NDE2NDM0OC81MzI5OTEwOTEtNmQ0ZTk5NWQtYjhjNi00ZDVmLWI2NjgtYTQ5MTE3ZTczYmJkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNjAxMDklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjYwMTA5VDAxMTAxOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTJhY2Y3MWViY2U2NzQxY2UxNjk4YmM5ZThjMjE0ZTY2NzFkNTE2ZjE1YzM4Y2QxM2E1YTA3NDM0ODA2YTVmZGMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.UIlVhcwjSop95_JFu1OVtpNXcKq9NIREBOiQW9wULf8"> Either remove or fluff the block chance description blurb

Ah, yeah, I was just using this to make sure everything was working properly. I'll cut it out

FalloutFalcon commented on code/game/objects/items/shields.dm:
```diff
@@ -77,164 +88,72 @@
-/obj/item/shield/riot/buckler
-	name = "wooden buckler"
+/obj/item/shield/buckler
+	name = "Wooden buckler"
```
this is not a proper noun
```suggestion
	name = "wooden buckler"
```

FalloutFalcon commented on code/game/objects/items/shields.dm:
```diff
@@ -77,164 +88,72 @@
-
 /obj/item/shield/energy
-	name = "energy combat shield"
+	name = "Energy combat shield"
```
ditto
```suggestion
	name = "energy combat shield"
```

FalloutFalcon commented on code/game/objects/items/shields.dm:
```diff
@@ -285,8 +205,8 @@
-/obj/item/shield/riot/tele
-	name = "telescopic shield"
+/obj/item/shield/tele
+	name = "Telescopic shield"
```
ditto
```suggestion
	name = "telescopic shield"
```

FalloutFalcon commented on code/game/objects/items/shields.dm:
```diff
@@ -326,18 +253,49 @@
-	lefthand_file = 'icons/mob/inhands/equipment/shields_lefthand.dmi'
-	righthand_file = 'icons/mob/inhands/equipment/shields_righthand.dmi'
+/obj/item/shield/heavy
+	name = "Heavy ballistic shield"
```
ditto
```suggestion
	name = "heavy ballistic shield"
```

FalloutFalcon commented on code/game/objects/items/shields.dm:
```diff
@@ -326,18 +253,49 @@
+
+	if(broken != TRUE)
+		block_chance = 60
+		slowdown = 1.75
```
your default slowdown is 1.5 so you created another weird mistake here
```suggestion
		block_chance = initial(block_chance)
		slowdown = initial(slowdown)
```

FalloutFalcon commented on code/game/objects/items/shields.dm:
```diff
@@ -326,18 +253,49 @@
+/obj/item/shield/heavy/proc/on_unwield(obj/item/source, mob/user)
+	SIGNAL_HANDLER
+
+	if(broken != TRUE)
```
nit:
```suggestion
	if(!broken)
```

FalloutFalcon commented on code/game/objects/items/shields.dm:
```diff
@@ -326,18 +253,49 @@
+/// triggered on wield of two handed item
+/obj/item/shield/heavy/proc/on_wield(obj/item/source, mob/user)
+
+	if(broken != TRUE)
```
nit:
```suggestion
	if(!broken)
```

FalloutFalcon commented on code/game/objects/items/shields.dm:
```diff
@@ -326,18 +253,49 @@
+/obj/item/shield/heavy/proc/on_wield(obj/item/source, mob/user)
+
+	if(broken != TRUE)
+		if(do_after(user, 30, user, IGNORE_USER_LOC_CHANGE | IGNORE_TARGET_LOC_CHANGE, TRUE))
```
```suggestion
		if(do_after(user, 3 SECONDS, user, IGNORE_USER_LOC_CHANGE | IGNORE_TARGET_LOC_CHANGE, TRUE))
```

FalloutFalcon commented on tools/UpdatePaths/Scripts/5505_shields_kill_repaths.txt:
```diff
@@ -0,0 +1,4 @@
+/obj/item/shield/riot : /obj/item/shield/heavy
+/obj/item/shield/riot/goliath : /obj/item/shield/buckler
+/obj/item/shield/riot/buckler : /obj/item/shield/buckler
+/obj/item/shield/riot/tele : /obj/item/shield/tele
```
prevents stuff like pixelshifting from being reset (which did infact happen with your pr rn and you may want to fix)
```suggestion
/obj/item/shield/riot : /obj/item/shield/heavy {@OLD}
/obj/item/shield/riot/goliath : /obj/item/shield/buckler {@OLD}
/obj/item/shield/riot/buckler : /obj/item/shield/buckler {@OLD}
/obj/item/shield/riot/tele : /obj/item/shield/tele {@OLD}
```

FalloutFalcon commented on code/game/objects/items/shields.dm:
```diff
@@ -77,164 +88,72 @@
+				broken = FALSE
+				block_chance = standard_block_chance
+				slowdown = standard_slowdown
+				drag_slowdown = standard_slowdown
```
I dont see anything that verifys you acctually HAVE 10 sheets

FalloutFalcon commented on code/game/objects/items/shields.dm:
```diff
@@ -77,164 +88,72 @@
 			return FALSE
 	. = ..()
-	if(.)
+	if(. && broken != TRUE)
```
```suggestion
	if(. && !broken)
```

FalloutFalcon commented on code/game/objects/items/shields.dm:
```diff
@@ -57,6 +67,7 @@
 			. += span_info("It appears heavily damaged.")
 		if(0 to 25)
 			. += span_warning("It's falling apart!")
+	. += span_info("Block chance is at [block_chance].")
```
```suggestion
	if(block_chance)
		. += span_info("It blocks [block_chance]% of attacks.")
```

FalloutFalcon reviewed (changes requested):
sorry for really late review

CluckinAwesome committed:
Fixes

Fixed repair function with referring to initial values alongside the unwield for heavy shields, made repair actually check for the amount required, fixed nouns alongside lots of nits, fixed (I think) the repath problem with resetting pixelshifting

CluckinAwesome committed:
Release me from this hell

CluckinAwesome commented:
I believe all changes have been made as requested, the bug still exists where the offhand is stuck if you interact while the do after is running, I'll continue trying to figure out how to resolve it

CluckinAwesome committed:
Merge branch 'master' into shieldUpdate

Signed-off-by: CluckinAwesome <158237114+CluckinAwesome@users.noreply.github.com>

generalthrax commented:
<img width="632" height="129" alt="image" src="https://github.com/user-attachments/assets/3a84ee62-0929-4335-9069-88102930f9c7" />
 stuffs fucked up

CluckinAwesome committed:
Merge branch 'master' into shieldUpdate

CluckinAwesome committed:
im tired boss

CluckinAwesome committed:
Merge branch 'master' into shieldUpdate

CluckinAwesome committed:
im tired boss

CluckinAwesome committed:
im really tired boss

CluckinAwesome committed:
Heavy Footsteps

fixed the wield bug, lost sp

CluckinAwesome commented:
All current issues should be resolved, as far as I'm aware, the PR is ready for re-reviews and TM again

FalloutFalcon commented:
<img width="1896" height="394" alt="image" src="https://github.com/user-attachments/assets/8e466b6c-baeb-4c8e-858b-ef2f71ad92e0" />

this prob only needs like one round to verify new changes works then get this merged.

rye-rice reviewed (approved):
nice sprites
